task8/TutterApp/src/main/java/dao/HashTagDAOImpl.java

```diff
@@ -0,0 +1,107 @@
+package dao;
+
+import models.HashTag;
+
+import javax.naming.NamingException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+
+public class HashTagDAOImpl implements HashTagDAO {
+    private static Logger logger;
+    private static ConnectionPool connectionPool;
+
+    static {
+        try {
+            LogManager.getLogManager().readConfiguration(HashTagDAOImpl.class.getClassLoader().getResourceAsStream("logging.properties"));
+
+            logger = Logger.getLogger(HashTagDAOImpl.class.getName());
+        } catch (Exception ignored) {
+        }
+    }
+
+    public HashTagDAOImpl() {
```

Comment:
This is a bad practice to initialize a static variable in an *instance* constructor. You'' break everything down if you have more than one instance... 
Actually, you do not need at all a `ConnectionPool` variable anywhere. Because the static method `ConnectionPool.getInstance()` will return you the one and only instance from any place of the code, any time